core/__tests__/tasks/recordProperty/enqueue/grouppedAggregations.ts

```diff
@@ -0,0 +1,134 @@
+          options: [],
+          methods: {
+            test: async () => {
+              return { success: true };
+            },
+          },
+        },
+      ],
+      connections: [
+        {
+          name: "test-connection",
+          displayName: "test-connection",
+          description: "a test connection",
+          apps: ["test-app"],
+          direction: "import",
+          options: [],
+          groupAggregations: [AggregationMethod.Exact],
+          methods: {
+            propertyOptions: async () => [
+              {
+                key: "aggregationMethod",
+                required: true,
+                description: "aggregationMethod",
+                type: "text",
+                options: async () => [
+                  { key: AggregationMethod.Exact, default: true },
+                ],
+              },
+            ],
+            uniquePropertyBootstrapOptions: async () => {
+              return { aggregationMethod: AggregationMethod.Exact };
+            },
+            sourceFilters: async () => [],
+            sourcePreview: async () => {
+              return [{ id: 1, email: "foo@example.com" }];
+            },
+            recordProperties: async () => {
+              // doesn't matter for this test
+              return {};
+            },
+          },
+        },
+      ],
+    });
+
+    const model = await helper.factories.model();
+
+    const app = await App.create({
+      type: "test-app",
+      name: "test app",
+    });
+    await app.update({ state: "ready" });
+
+    const source = await Source.create({
+      type: "test-connection",
+      name: "test connection",
+      appId: app.id,
+      modelId: model.id,
+    });
+    userId = await source.bootstrapUniqueProperty({ mappedColumn: "id" });
+    await source.setMapping({ id: "profiles_id" });
+    await source.update({ state: "ready" });
+
+    email = await helper.factories.property(
+      source,
+      { key: "email" },
+      { aggregationMethod: AggregationMethod.Exact }
+    );
+    await email.update({ state: "ready" });
+  });
+
+  describe("recordProperties:enqueue", () => {
+    test("large batches will be broken up", async () => {
+      process.env.GROUPAROO_INTERNAL_BATCH_SIZE = "2";
+      rebuildConfig();
+
+      const mario = await helper.factories.record();
+      const luigi = await helper.factories.record();
+      const toad = await helper.factories.record();
+      const peach = await helper.factories.record();
+
+      await mario.markPending();
+      await luigi.markPending();
+      await toad.markPending();
+      await peach.markPending();
+
+      // shuffle the order of things so all the updated properties will be found in the first group
+      for (const person of [mario, luigi, toad, peach]) {
+        await api.sequelize.query(
+          `UPDATE "recordProperties" SET "updatedAt"=NOW() WHERE "recordId"='${person.id}' and "propertyId"='${email.id}'`
+        );
```

Comment:
Curious why not use the ORM for this

Reply:
74e28410f98e9ad49f3e250a57059d5e4ef0914e makes things a little nicer